File: frontend/src/types/timeline.ts
export interface TimelineEvent {
  id: string;
  record_type: string;
  display_text: string;
  effective_date: string | null;
  code_display: string | null;
  category: string[] | null;
  // Human-readable provider/performer, server-derived from the record's
  // fhir_resource. Null when the record carries none.
  provider?: string | null;
}

export interface TimelineResponse {
  events: TimelineEvent[];
  total: number;
}

export interface TimelineStats {
  total_records: number;
  records_by_type: Record<string, number>;
  date_range_start: string | null;
  date_range_end: string | null;
}

File: backend/app/schemas/timeline.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class TimelineEvent(BaseModel):
    id: UUID
    record_type: str
    display_text: str
    effective_date: datetime | None
    code_display: str | None
    category: list[str] | None
    provider: str | None = None

    model_config = {"from_attributes": True}


class TimelineResponse(BaseModel):
    events: list[TimelineEvent]
    total: int


class TimelineStats(BaseModel):
    total_records: int
    records_by_type: dict[str, int]
    date_range_start: datetime | None
    date_range_end: datetime | None
